src/libcharon/plugins/kernel_super/oper_db_client/liboper_db_client/liboper_db_client.h

```diff
@@ -47,6 +102,12 @@ void liboper_db_client_db_obj_destroy(liboper_db_obj_t* obj);
 
 bool liboper_db_client_db_obj_enum_set(liboper_db_obj_t* obj, const enum LIBOPER_DB_CLIENT_FIELD_IDS field_id, enum LIBOPER_DB_CLIENT_FIELD_ENUM_VALUES enum_value);
 bool liboper_db_client_db_obj_u64_set(liboper_db_obj_t* obj, const enum LIBOPER_DB_CLIENT_FIELD_IDS field_id, uint64_t new_num);
+bool liboper_db_client_db_obj_u16_set(liboper_db_obj_t* obj, const enum LIBOPER_DB_CLIENT_FIELD_IDS field_id, uint16_t new_num);
+bool liboper_db_client_db_obj_u8_set(liboper_db_obj_t* obj, const enum LIBOPER_DB_CLIENT_FIELD_IDS field_id, uint8_t new_num);
+bool liboper_db_client_db_obj_bool_set(liboper_db_obj_t* obj, const enum LIBOPER_DB_CLIENT_FIELD_IDS field_id, bool new_val);
+bool liboper_db_client_db_obj_string_set(liboper_db_obj_t* obj, const enum LIBOPER_DB_CLIENT_FIELD_IDS field_id, const char *new_data);
+bool liboper_db_client_db_obj_union_str_set(liboper_db_obj_t* obj, const enum LIBOPER_DB_CLIENT_FIELD_IDS field_id, struct liboper_union *str_val);
```

Comment:
Shouldn't use `struct liboper_union` as it contains other fields, so using this struct doesn't really help with the reason of splitting `union_set()` to union_str_set()` - For example it's possible for the user to set the `double_num` field in `struct liboper_union` and call `union_str_set()` by mistake.

Instead split the `struct liboper_union` to separate structs for each field (`struct liboper_union_str` in our case) with only the needed fields (`const char*` and `field_id` for the string case)